pkg/api/deprovision.go

```diff
@@ -142,21 +143,46 @@ func (s *server) deprovision(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	task := model.NewTask(
-		"deprovisionStep",
-		map[string]string{
-			"stepName":   firstStepName,
-			"instanceID": instanceID,
-		},
-	)
+	childCount, err := s.store.GetInstanceChildCountByAlias(instance.Alias)
+	if err != nil {
+		logFields["step"] = firstStepName
+		logFields["error"] = err
+		log.WithFields(logFields).Error(
+			"deprovisioning error: error determining child count",
+		)
+		s.writeResponse(w, http.StatusInternalServerError, responseEmptyJSON)
+	}
+
+	var task model.Task
+	if childCount > 0 {
+		task = model.NewDelayedTask(
+			"waitForChildrenStep",
```

Comment:
Semantics: existing task names are things like `deprovisionStep` (emphasis on _step_) because that's what those tasks do-- they execute one _step_ of the deprovisioning process. (Step is a "technical" term here, in a manner of speaking, because the provisioners and deprovisioners are literally composed of "steps.") I'm not sure how applicable "step" is to what we're doing here. `waitForChildren` (sans "step"), might be a more appropriate task name.

Part 2-- more semantics: `waitForChildren` still might not adequately describe the task. The task doesn't do much waiting. It's awake for a moment before either scheduling another task like itself or scheduling the first deprovisioning step. Maybe `checkChildrensStatuses` works?

Sorry to split hairs so much, but the naming of tasks (for instance) does fall into the category of things that we eventually can not change anymore, so I want to get it right.

Reply:
👍